### DISCUSSION TITLE
Separate Highlight Groups for Different Languages.

### DISCUSSION THREAD

**Discussion by CREESTL** [rose-pine/neovim#117] [Q&A] [answered]
Currently I've overridden `highlight_groups` for one language (Solidity). But. of course, it also changed colors for *all other languages*.   
I want to use specific `highlight_groups` for a single language and use dafault ones for other languages.   
I want something like this:
```
if (language == "Solidity") {

  highlight_groups = {
    ColorColumn = { bg = 'rose' },
    ['@constructor'] = { fg = 'gold', style = "bold" },
    ['@keyword'] = { fg = 'gold', style = "bold" },  
   ...        
   }

}
```	
Is it possible?  

I'm using TreeSitter with [Solidity](https://github.com/YongJieYongJie/tree-sitter-solidity) parser installed

**Comment by mvllow** [comment, marked as answer]
This is out of the scope of the theme at the moment but perhaps you could manually override colours via an autocmd:

```lua
vim.api.nvim_create_autocmd("ColorScheme", {
  pattern = ".solidity", -- Or whatever the file extension is
  callback = function()
    -- This hardcodes our "gold" colour but you could require our palette as well.
    vim.api.nvim_set_hl(0, "@constructor", { fg = "#f6c177", style = "bold" })
    vim.api.nvim_set_hl(0, "@keyword", { fg = "#f6c177", style = "bold" })
  end,
})
```

**Reply by CREESTL** [reply]
Yeah, the `autocmd` worked for me. But the `ColorScheme` event had no matches for `.sol` files. So I used the `FileType` instead.  Also `style = "bold"` doesn't work for Nvim Lua API. I used `bold=true` for that.
Here's what I have:  
```
    vim.api.nvim_create_autocmd("FileType", {
        pattern = {"solidity"}, 
        callback = function()
            vim.api.nvim_set_hl(0, "@constructor", { fg = "#f6c177", bold=true})
            vim.api.nvim_set_hl(0, "@keyword", { fg = "#f6c177" })
        end,
    })
```
